### PR TITLE
LPD-42480 Fix DispatcherType and path handling for JBoss

### PR DESCRIPTION
**Pull request by dannielraposo**
- **Problem?**

There is a bug in `RESTClientTemplateContextContributor` that affects the creation of `RESTClientHttpRequest` objects on JBoss servers. When wrapping the original HTTP request, the `getDispatcherType` method does not correctly retrieve the dispatcher type from the `RequestDispatcher`, causing it to default to `INCLUDE` instead of the intended `FORWARD`. Additionally, the request path is not properly obtained from the `RequestDispatcher`, resulting in a missing path. These issues prevent the HTTP request from reaching the intended REST endpoint correctly.

- **Changes?**

The `DispatcherType` has been defaulted to `FORWARD`, and the request path is now passed as a parameter when creating the `RESTClientHttpRequest`.

See the following Jira Ticket: [LPD-42480](https://liferay.atlassian.net/browse/LPD-42480)

### CODE REVIEW

**Comment by dannielraposo** [comment]
ci:test:relevant

File: modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/template/servlet/RESTClientHttpRequest.java [hunk 1]
```diff
@@ -48,9 +48,11 @@
 public class RESTClientHttpRequest implements HttpServletRequest {
 
 	public RESTClientHttpRequest(
-		Map<String, Object> contextObjects,
+		String path, Map<String, Object> contextObjects,
```
Comment (4lejandrito) [inline modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/template/servlet/RESTClientHttpRequest.java:51]: Sort the parameters alphabetically (path should be the last one).

**Review by 4lejandrito — CHANGES_REQUESTED** [review]

File: modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/template/servlet/RESTClientHttpRequest.java [hunk 2]
```diff
@@ -277,7 +279,7 @@ public Collection<Part> getParts() throws IOException, ServletException {
 
 	@Override
 	public String getPathInfo() {
-		return _httpServletRequest.getPathInfo();
+		return _path;
```
Comment (ccorreagg) [inline modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/template/servlet/RESTClientHttpRequest.java:282]: I guess for the previous case (Tomcat) _path and pathInfo contained the same info, right?

File: modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/template/servlet/RESTClientHttpRequest.java [hunk 3]
```diff
@@ -48,9 +48,11 @@
 public class RESTClientHttpRequest implements HttpServletRequest {
 
 	public RESTClientHttpRequest(
-		Map<String, Object> contextObjects,
+		String path, Map<String, Object> contextObjects,
 		HttpServletRequest httpServletRequest) {
 
+		_path = path;
```
Comment (ccorreagg) [inline modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/template/servlet/RESTClientHttpRequest.java:54]: Same here with the order

**Review by ccorreagg — COMMENTED** [review]
This PR is magic

**Review by 4lejandrito — APPROVED** [review]
@dannielraposo please verify it still works on master with tomcat. If so, please feel free to forward.

**Comment by dannielraposo** [comment]
> @dannielraposo please verify it still works on master with tomcat. If so, please feel free to forward.

verified, it works.

**Comment by dannielraposo** [comment]
ci:forward

**Pull request closed without merging**